src/Mono.Android/Android.Runtime/AndroidRuntime.cs

```diff
@@ -530,20 +558,78 @@ public void RegisterNativeMembers (JniType nativeClass, Type type, ReadOnlySpan<
 							while (callbackDeclaringType.ContainsGenericParameters) {
 								callbackDeclaringType = callbackDeclaringType.BaseType!;
 							}
-							GetCallbackHandler connector = (GetCallbackHandler) Delegate.CreateDelegate (typeof (GetCallbackHandler),
-								callbackDeclaringType, callbackString.ToString ());
-							callback = connector ();
+#if ENABLE_MARSHAL_METHODS
+							// TODO: this is temporary hack, it needs a full fledged registration mechanism for methods like these (that is, ones which
+							// aren't registered with [Register] but are baked into Mono.Android's managed and Java code)
+							bool createCallback = false;
+							string declaringTypeName = callbackDeclaringType.FullName;
+							string callbackName = callbackString.ToString ();
+
+							foreach (var kvp in dynamicRegistrationMethods) {
+								string dynamicTypeName = kvp.Key;
+
+								foreach (string dynamicCallbackMethodName in kvp.Value) {
+									if (ShouldRegisterDynamically (declaringTypeName, callbackName, dynamicTypeName, dynamicCallbackMethodName)) {
+										createCallback = true;
+										break;
+									}
+								}
+
+								if (createCallback) {
```

Comment:
This raises a "meta-question": the implication here is that whether we register via `JNIEnv::RegisterNatives()` or via C `Java_…` function is done on a *per-method* basis.

Is that actually supported on Android?

Rephrased; given the Java type:

```java
package example;

public class C {
    public static native void m_viaRegister();
    public static native void m_viaFunction();
}
```

and then at runtime we have a `Java_example_C_m_1viaFunction()` function in a loaded `.so`, and we also call `JNIEnv:RegisterNatives()` but *only* for `m_viaRegister()`…

Does that work?  The JNI docs are not clear on how these two mechanisms interact.

For that matter, if *both* native mechanisms are provided, which mechanism actually invoked at runtime?  Which has priority?

Reply:
Answering that would require a deeper dive into Android JNI sources I suppose...

Reply:
I don't think this requires reading Android source.  An on-device test would answer this as well.  (Though that answer could change over time, and thus wouldn't be a "final" answer, but it could still be informative.)